### PR TITLE
docs: add file uploads

### PR DESCRIPTION
**Pull request by abernix**
Reverts apollographql/router#5167, which reverted the original https://github.com/apollographql/router/pull/4626 by @Meschreiber — I do actually think the failing tests are a result of something _in_ the PR, in the form of docs.  This should be re-analyzed!

### CODE REVIEW

File: docs/source/executing-operations/file-uploads.mdx [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+```yaml title="router.yaml" showLineNumbers="false"
+preview_file_uploads:
+  enabled: true
+  protocols:
+    multipart:
+      enabled: true
+      mode: stream
+      limits:
+        max_file_size: 1mb
+        max_files: 10
```
Comment (Meschreiber) [inline docs/source/executing-operations/file-uploads.mdx:92]: This `router.yaml` appears to be causing [this error](https://app.circleci.com/pipelines/github/apollographql/router/21502/workflows/c71c8bba-2fc0-4501-b86f-d88b98cf7976/jobs/147099)

```
--- STDERR:              apollo-router configuration::tests::validate_project_config_files ---
thread 'configuration::tests::validate_project_config_files' panicked at apollo-router/src/configuration/tests.rs:456:21:
../docs/source/executing-operations/file-uploads.mdx configuration error: 
failed to parse yaml: mapping values are not allowed in this context at line 2 column 21
```

However, this config matches what I could find in the `[default.router.yaml](https://github.com/apollographql/router/blob/99ad27a2f749d5d3a311014a2dc215cdfde7b240/apollo-router/tests/fixtures/file_upload/default.router.yaml#L4)` and `[file_uploads.router.yaml](https://github.com/apollographql/router/blob/99ad27a2f749d5d3a311014a2dc215cdfde7b240/apollo-router/src/configuration/testdata/metrics/file_uploads.router.yaml#L4)` exemplars.

**Review by Meschreiber — APPROVED** [review]

File: docs/source/executing-operations/file-uploads.mdx [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+By default, receiving client file uploads isn't enabled in the Apollo Router.
+To enable file upload support, set the following fields in your `router.yaml` configuration file:
+
+```yaml title="router.yaml"
+preview_file_uploads:
+  enabled: true
+  protocols:
+    multipart:
+      enabled: true
+      mode: stream
```
Comment (abernix) [inline docs/source/executing-operations/file-uploads.mdx:89]: I will note that this `mode: stream` bit has differed from the original PR — https://github.com/apollographql/router/compare/docs/file-uploads..revert-5167-revert-4626-docs/file-uploads

File: docs/source/executing-operations/file-uploads.mdx [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+#### Configuration reference
+
+The following are attributes of the root [`preview_file_uploads`](#configure-file-upload-support-in-the-router) configuration.
+
+<table class="field-table">
+<thead>
+<tr>
+<th>Attribute/ <br/> Description</th>
+<th>Default value</th>
+<th>Valid Values</th>
```
Comment (bnjjj) [inline docs/source/executing-operations/file-uploads.mdx:138]:
```suggestion
<th>Valid values</th>
```

**Review by bnjjj — APPROVED** [review]
1 nit, otherwise LGTM

File: docs/source/executing-operations/file-uploads.mdx (hunk 3 above)
Comment (abernix) [reply docs/source/executing-operations/file-uploads.mdx:138]: I might even turn that around and say that `Default value` is the thing that should be `Default Value`. ;)

File: docs/source/executing-operations/file-uploads.mdx [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+
+#### Configuration reference
+
+The following are attributes of the root [`preview_file_uploads`](#configure-file-upload-support-in-the-router) configuration.
+
+<table class="field-table">
+<thead>
+<tr>
+<th>Attribute/ <br/> Description</th>
+<th>Default value</th>
```
Comment (Meschreiber) [inline docs/source/executing-operations/file-uploads.mdx:137]:
```suggestion
<th>Default Value</th>
```

**Pull request merged**